scala-package/core/src/main/scala/org/apache/mxnet/NDArray.scala

```diff
@@ -64,22 +64,36 @@ object NDArray {
     val function = functions(funcName)
     val ndArgs = ArrayBuffer.empty[NDArray]
     val posArgs = ArrayBuffer.empty[String]
-    args.foreach {
-      case arr: NDArray =>
-        ndArgs.append(arr)
-      case arrFunRet: NDArrayFuncReturn =>
-        arrFunRet.arr.foreach(ndArgs.append(_))
-      case arg =>
-        posArgs.append(arg.toString)
+    if (args != null) {
+      args.foreach {
+        case arr: NDArray =>
+          ndArgs.append(arr)
+        case arrFunRet: NDArrayFuncReturn =>
+          arrFunRet.arr.foreach(ndArgs.append(_))
+        case arg =>
+          posArgs.append(arg.toString)
+      }
     }
-
     require(posArgs.length <= function.arguments.length,
       s"len(posArgs) = ${posArgs.length}, should be less or equal to len(arguments) " +
       s"= ${function.arguments.length}")
     val updatedKwargs: Map[String, String] =
       (Option(kwargs).getOrElse(Map.empty[String, String])
         ++ function.arguments.slice(0, posArgs.length).zip(posArgs) - "out"
-      ).map { case (k, v) => k -> v.toString }
+      ).filter{ case (key, value) =>
+        !value.isInstanceOf[NDArray] && !value.isInstanceOf[NDArrayFuncReturn]
```

Comment:
Why do we need to filter this now? Are we getting any new values?

Reply:
For new API, NDarray are mixed in the kwargs, we don’t want these to become string... instead to store into a ndArgs list